Cannot build atomics example with LTO enabled

...and I really cannot explain why, and no matter what I try, I cannot get it to link properly. 

The example located at `kos/examples/dreamcast/basic/threading/atomics` fails to link with LTO enabled. This is a known issue:
```
/opt/toolchains/dc/sh-elf-13.2.0/bin/../lib/gcc/sh-elf/13.2.0/../../../../sh-elf/bin/ld: /tmp/cc173KsQ.ltrans0.ltrans.o: in function `_atomic_add_buffer':
/opt/toolchains/dc/kos/examples/dreamcast/basic/threading/atomics/atomics.c:109:(.text.atomic_add_buffer+0xd4): undefined reference to `___atomic_load'
/opt/toolchains/dc/sh-elf-13.2.0/bin/../lib/gcc/sh-elf/13.2.0/../../../../sh-elf/bin/ld: /opt/toolchains/dc/kos/examples/dreamcast/basic/threading/atomics/atomics.c:109:(.text.atomic_add_buffer+0xe8): undefined reference to `___atomic_compare_exchange'
/opt/toolchains/dc/sh-elf-13.2.0/bin/../lib/gcc/sh-elf/13.2.0/../../../../sh-elf/bin/ld: /tmp/cc173KsQ.ltrans0.ltrans.o: in function `_thread.lto_priv.0':
/opt/toolchains/dc/kos/examples/dreamcast/basic/threading/atomics/atomics.c:121:(.text.thread.lto_priv.0+0x230): undefined reference to `___atomic_fetch_add_8'
/opt/toolchains/dc/sh-elf-13.2.0/bin/../lib/gcc/sh-elf/13.2.0/../../../../sh-elf/bin/ld: /tmp/cc173KsQ.ltrans1.ltrans.o: in function `main.constprop.0':
/opt/toolchains/dc/kos/examples/dreamcast/basic/threading/atomics/atomics.c:266:(.text.main.constprop.0+0x158): undefined reference to `___atomic_is_lock_free'
/opt/toolchains/dc/sh-elf-13.2.0/bin/../lib/gcc/sh-elf/13.2.0/../../../../sh-elf/bin/ld: /opt/toolchains/dc/kos/examples/dreamcast/basic/threading/atomics/atomics.c:325:(.text.main.constprop.0+0x400): undefined reference to `___atomic_load_8'
/opt/toolchains/dc/sh-elf-13.2.0/bin/../lib/gcc/sh-elf/13.2.0/../../../../sh-elf/bin/ld: /opt/toolchains/dc/kos/examples/dreamcast/basic/threading/atomics/atomics.c:325:(.text.main.constprop.0+0x414): undefined reference to `___atomic_load'
collect2: error: ld returned 1 exit status
make: *** [Makefile:22: atomics.elf] Error 1
```

If anyone has any ideas, I'm all ears. This one has stumped me. I'm worried there's some fundamental incompatibility between LTO and the pseudo compiler-intrinsics that resolve to KOS's atomics back-end at link-time... No idea.

--- a/kernel/arch/dreamcast/kernel/init.c
+++ b/kernel/arch/dreamcast/kernel/init.c
@@ -388,6 +388,7 @@ void arch_menu(void) {
 
 /* Called to shut down non-gracefully; assume the system is in peril
    and don't try to call the dtors */
+__used __noreturn
 void arch_abort(void) {
     /* Disable the WDT, if active */
     wdt_disable();

--- a/kernel/thread/thread.c
+++ b/kernel/thread/thread.c
@@ -828,6 +828,7 @@ void thd_sleep(unsigned int ms) {
 }
 
 /* Manually cause a re-schedule */
+__used
 void thd_pass(void) {
     /* Makes no sense inside int */
     if(irq_inside_int()) return;

